content/en/docs/reference/helper-libraries/dra-driver-controller.md

```diff
@@ -0,0 +1,112 @@
+---
+title: DRA Resource Driver controller helper library
+description: Handles most of functionality common for all DRA resource drivers' controllers.
+weight: 10
+---
+
+<!-- overview -->
+
+When implementing
+[Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) (DRA)
+resource driver controller, the communication with the scheduler is done through API objects.
+
+This library implements some of the functionality that is common for all resource
+drivers, regardless of the actual resource managed or allocation logic. This includes, but isn't restricted to:
+
+- watching and caching of {{< glossary_tooltip term_id="resource-claim" text="ResourceClaims">}}
+- updating the ResourceClaim when the allocation or deallocation is done
+- watching and caching of {{< glossary_tooltip term_id="pod-scheduling-context" text="PodSchedulingContext">}}
+- updating the PodSchedulingContext
+- registering with Kubernetes cluster to handle particular {{< glossary_tooltip term_id="resource-class" text="ResourceClass">}}
+
+<!-- body -->
+
+## Usage
+
+Controller helper library communicates with the scheduler through ResourceClaim
+and PodSchedulingContext objects. The 3rd party implementation of the resource
+driver only needs to implement the validation of the ResourceClaim and
+ResourceClass parameters objects, and the allocation logic itself.
+
+The controller helper library is expecting the actual resource driver controller
+to implement following interface:
```

Comment:
I prefer to not duplicate this API documentation here.

Let's instead link to https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.29.0/controller#Controller

Reply:
If we do that, we'd also want to ensure that the release docs team update the link each release.

For that and other reasons, I think it's more useful to put the details here. People who aren't contributing to Kubernetes would find this information useful.

Reply:
> If we do that, we'd also want to ensure that the release docs team update the link each release.

We can also link to https://pkg.go.dev/k8s.io/dynamic-resource-allocation/controller#Driver

There's a (small) risk that the link might go stale, but that's not different from copying the documentation here and not updating it.

> For that and other reasons, I think it's more useful to put the details here. 

But not at this level of details. Developers who want to write a DRA drivers should consult the package documentation, not this page. There's much more information there that they should read.

> People who aren't contributing to Kubernetes would find this information useful.

A summary yes, but all the details?

Reply:
Replaced the interface quote with a link to _latest_ package version API reference. Only when the interface changes, then the documentation will need to be updated. A compromise.